### PR TITLE
gh-116267: Do not check record size in RotatingFileHandler to avoid f…

### PR DESCRIPTION
**Pull request by Jason-Y-Z**
…ormatting twice

This stops us from formatting the new record twice in `RotatingFileHandler`.

* Issue: gh-116267

### CODE REVIEW

File: Doc/library/logging.handlers.rst [hunk 1]
```diff
@@ -342,6 +342,10 @@ module, supports rotation of disk log files.
    .. versionchanged:: 3.9
       The *errors* parameter was added.
 
+   .. versionchanged:: 3.13
```
Comment (serhiy-storchaka) [inline Doc/library/logging.handlers.rst]:
```suggestion
   .. versionchanged:: 3.14
```

**Comment by serhiy-storchaka** [comment]
@vsajip, what are your thoughts about this?

File: Lib/test/test_logging.py [hunk 2]
```diff
@@ -6148,6 +6148,8 @@ def test_should_not_rollover(self):
 
     def test_should_rollover(self):
         rh = logging.handlers.RotatingFileHandler(self.fn, encoding="utf-8", maxBytes=1)
+        self.assertFalse(rh.shouldRollover(self.next_rec()))
+        rh.emit(self.next_rec())
```
Comment (serhiy-storchaka) [inline Lib/test/test_logging.py:6152]: `self.next_rec()` returns a new record. You should save it and pass the same record to `shouldRollover()` and `emit()`.

**Comment by vsajip** [comment]
I'm not sure why @Jason-Y-Z closed this - maybe due to impatience due to my lack of response. I've not had much time to look at this, but I was thinking about trying a different approach - caching the formatted value in an instance variable to avoid reformatting. Since everything happens inside `emit()`, in theory one could avoid coupling by replacing the `FileHandler.emit(...)` call with just the `stream.write(...)` and `self.flush()` that `StreamHandler.emit()` does (since `self.stream` won't be `None` at that point, `FileHandler.emit()` is essentially the same as `StreamHandler.emit(...)` - it's just two lines of code duplication, and so doesn't seem unreasonable to do to avoid coupling. Thoughts?

**Pull request closed without merging**